### PR TITLE
feat: added support for the sdk roles contract instead of custom roles definitions

### PR DESCRIPTION
**Pull request by Foivos**
Also added a mint limiter role for token managers
[AXE-1733]

[AXE-1733]: https://axelarnetwork.atlassian.net/browse/AXE-1733?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/utils/Distributable.sol [hunk 1]
```diff
@@ -4,80 +4,63 @@ pragma solidity ^0.8.0;
...
      */
-    function proposeDistributorship(address distributor_) external onlyDistributor {
-        assembly {
-            sstore(PROPOSED_DISTRIBUTOR_SLOT, distributor_)
-        }
-        emit DistributorshipTransferStarted(distributor_);
+    function proposeDistributorship(address distributor_) external onlyRole(uint8(Roles.DISTRIBUTOR)) {
+        uint8[] memory roles = new uint8[](1);
+        roles[0] = uint8(Roles.DISTRIBUTOR);
+        _proposeRoles(msg.sender, distributor_, roles);
```
[the remark this answers is not in the document]
Comment (Foivos) [reply contracts/utils/Distributable.sol:46]: yea maybe worthwhile, same with _transferRoles.

File: contracts/test/HardCodedConstantsTest.sol [hunk 2]
```diff
@@ -46,10 +43,7 @@ contract TestNoReEntrancy is NoReEntrancy {
 contract TestOperatable is Operatable {
```
Comment (milapsheth) [inline contracts/test/HardCodedConstantsTest.sol:43]: Test can be deleted I guess

File: contracts/utils/Operatable.sol [hunk 3]
```diff
@@ -4,79 +4,57 @@ pragma solidity ^0.8.0;
...
      * @param operator_ The address of the new operator
      */
-    function _setOperator(address operator_) internal {
-        assembly {
-            sstore(OPERATOR_SLOT, operator_)
-        }
-        emit OperatorshipTransferred(operator_);
+    function _addOperator(address operator_) internal {
+        uint8[] memory roles = new uint8[](1);
+        roles[0] = uint8(Roles.OPERATOR);
```
Comment (milapsheth) [inline contracts/utils/Operatable.sol:23]: Use add role here now

**Review by milapsheth — APPROVED** [review]

File: contracts/test/HardCodedConstantsTest.sol (hunk 2 above)
Comment (Foivos) [reply contracts/test/HardCodedConstantsTest.sol:43]: no it cannot, it is used in testing

File: contracts/utils/Operatable.sol (hunk 3 above)
Comment (Foivos) [reply contracts/utils/Operatable.sol]: fixed

**Pull request merged**